Dockerfile

```diff
@@ -48,8 +49,15 @@ RUN docker-php-ext-install   pcntl \
         bcmath \
         calendar \
         pdo_mysql \
+        sockets \
         zip
 
+ENV GET_COMPOSER_VERSION="76a7060ccb93902cd7576b67264ad91c8a2700e2"
```

Comment:
the image already includes the vendor with packages installed, i was using this method, didn't see any advantages for it honestly, unless you have a good point why this method is better then i'll accept it.

it will just slow the build time while it's easier to just copy the vendor to the image.

Reply:
My concern is to be able to reproduce the Docker image locally, so that I don't have to rely on the binary provided by Dockerhub.

If the image on Dockerhub already contains the relevant dependencies, it means that they had to be installed manually locally _at some point_, right?
In that case my concern is: how can other people reproduce the same image locally so that the resulting image is identical to the Dockerhub version?

With this change the time it takes to build the Docker image itself actually won't be that much longer, since composer is downloaded from the binary releases (and the subsequent builds will be just as fast as before, once the Docker cache is populated).

What will take significantly more time though, is at the container runtime,  to run `composer` install from the entrypoint to install the actual dependencies.
IMO if that is an issue, the right approach would be to change it so that the dependencies themselves can also be bundled in the Docker image itself (i.e. install the dependencies from the Dockerfile instead of from the entrypoint).
That way it is possible to write the Dockerfile in such a way that the Docker cache is always used (which means that rebuilding that part of the image takes less than 1 second) if no dependency changed.
When I tried it this was not possible because it complained about the SQLite database not being present, so it required much more changes.